src/jobservice/pool/message_server.go

```diff
@@ -155,6 +200,22 @@ func (ms *MessageServer) Subscribe(event string, callback interface{}) error {
 	return nil
 }
 
+func (ms *MessageServer) shouldRetry() bool {
+	flag := ms.retryFlag.Load().(bool)
+	count := ms.retryCounter.Load().(int)
+	yesOrNo := flag && count < retryTimes
+	if yesOrNo {
+		count++
```

Comment:
If the func `Start()` is not thread-safe, `shouldRetry()` is not, so then `count++` is problematic.

Reply:
Right. Should use safe increase method